src/Compilers/Core/Portable/EncodedStringText.cs

```diff
@@ -8,7 +8,7 @@
 
 namespace Microsoft.CodeAnalysis.Text
 {
-    internal static class EncodedStringText
+    public static class EncodedStringText
```

Comment:
I like the idea of a "detectEncoding" parameter for one of the existing factory methods.  I'll work on that and resend.